src/execution/index.ts

```diff
@@ -293,14 +301,62 @@ async function waitForExecution(
   switch (result.type) {
     case 'intent': {
       let intentStatus: IntentOpStatus | null = null
+      const startTs = Date.now()
+      let nextDelayMs = POLL_INITIAL_MS
+      let errorBackoffMs = POLL_ERROR_BACKOFF_MS
       while (intentStatus === null || !validStatuses.has(intentStatus.status)) {
+        const now = Date.now()
+        if (now - startTs >= POLL_MAX_WAIT_MS) {
+          throw new IntentStatusTimeoutError({
+            context: { waitedMs: now - startTs },
+          })
+        }
         const orchestrator = getOrchestratorByChain(
           result.targetChain,
           config.apiKey,
           config.endpointUrl,
         )
-        intentStatus = await orchestrator.getIntentOpStatus(result.id)
-        await new Promise((resolve) => setTimeout(resolve, POLLING_INTERVAL))
+        try {
+          intentStatus = await orchestrator.getIntentOpStatus(result.id)
+          // reset error backoff on success
+          errorBackoffMs = POLL_ERROR_BACKOFF_MS
```

Comment:
so to recap, this will try every 500ms for the first 5s and then every 2s for 3min in total right?